libcloud/dns/drivers/auroradns.py

```diff
@@ -222,6 +358,205 @@ def update_record(self, record, name, type, data, extra):
 
         return self.get_record(record.zone.id, record.id)
 
+    def ex_list_healthchecks(self, zone):
+        """
+        List all Health Checks in a zone.
+
+        :param zone: Zone to list health checks for.
+        :type zone: :class:`Zone`
+
+        :return: ``list`` of :class:`AuroraDNSHealthCheck`
+        """
+        healthchecks = []
+        self.connection.set_context({'resource': 'zone', 'id': zone.id})
+        res = self.connection.request('/zones/%s/health_checks' % zone.id)
+
+        for healthcheck in res.parse_body():
+            healthchecks.append(self.__res_to_healthcheck(zone, healthcheck))
+
+        return healthchecks
+
+    def ex_get_healthcheck(self, zone, health_check_id):
+        """
+        Get a single Health Check from a zone
+
+        :param zone: Zone in which the health check is
+        :type zone: :class:`Zone`
+
+        :param health_check_id: ID of the required health check
+        :type  health_check_id: ``str``
+
+        :return: :class:`AuroraDNSHealthCheck`
+        """
+        self.connection.set_context({'resource': 'healthcheck',
+                                     'id': health_check_id})
+        res = self.connection.request('/zones/%s/health_checks/%s'
+                                      % (zone.id, health_check_id))
+        check = res.parse_body()
+
+        return self.__res_to_healthcheck(zone, check)
+
+    def ex_create_healthcheck(self, zone, type, hostname, port, path,
+                              interval, threshold, ipaddress=None,
+                              enabled=True, extra=None):
+        """
+        Create a new Health Check in a zone
+
+        :param zone: Zone in which the health check should be created
+        :type zone: :class:`Zone`
+
+        :param type: The type of health check to be created
+        :type  type: :class:`AuroraDNSHealthCheckType`
+
+        :param hostname: The hostname of the target to monitor
+        :type  hostname: ``str``
+
+        :param port: The port of the target to monitor. E.g. 80 for HTTP
+        :type  port: ``int``
+
+        :param path: The path of the target to monitor. Only used by HTTP
+                     at this moment. Usually this is simple /.
+        :type  path: ``str``
+
+        :param interval: The interval of checks. 10, 30 or 60 seconds.
+        :type  interval: ``int``
+
+        :param threshold: The threshold of failures before the healthcheck is
+                          marked as failed.
+        :type  threshold: ``int``
+
+        :param ipaddress: (optional) The IP Address of the target to monitor.
+                          You can pass a empty string if this is not required.
+        :type  ipaddress: ``str``
+
+        :param enabled: (optional) If this healthcheck is enabled to run
+        :type  enabled: ``bool``
+
+        :param extra: (optional) Extra attributes (driver specific).
```

Comment:
```extra``` doesn't seem to be used.  Is it needed?  Are there other 'health check' drivers that use it and this is just allowing it to be passed in for consistency?